src/routes/PaymentCheckPage.tsx

```diff
@@ -123,8 +123,9 @@ export default function PaymentCheckPage() {
     window.location.replace(authorizationUrl);
   };
 
-  // TODO CHK-923
-  const [month, year] = cardData.expDate.split("/");
+  const stDate = "01/".concat(cardData?.expDate || "");
+  const pattern = /(\d{2})\/(\d{2})\/(\d{2})/;
+  const dateParsed = new Date(stDate.replace(pattern, "$2-$1-$3"));
```

Comment:
This is a very elegant solution but it seems that our technology stack is a little bit old to support this notation.
Target in ts config need to be set to es6 and at runtime i get this error T`ypeError: String.prototype.matchAll called with a non-global RegExp argument`. Any suggestion?

Reply:
It seems that `matchAll` requires the `global` flag (else it throws a `TypeError`, see [here](https://stackoverflow.com/questions/60289995/matchall-throws-error-when-g-flag-is-missing-now) and, obviously missed by me, [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll) 😅.
In our case we expect one match only, so we can either add the `g` flag or use just [`match`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match) instead (without `g`, see docs)

```typescript
const [date, month, year] = stDate.match(regex).splice(1)
```

Reply:
Set in this way with to edit:

```
const datePattern = /(\d{2})\/(\d{2})\/(\d{2})/;
const [day, month, year] = stDate.match(regex).slice(1);
const dateParsed = new Date(Number(year)+2000, Number(month) - 1, Number(day));
```

first edit `slice` method in place of splice since it effectively return an array as expected
second edit in date constructor add 2000 to the year, since from documentation [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date) `year` parameter in date constructor is _Integer value representing the year. **Values from 0 to 99 map to the years 1900 to 1999**. All other values are the actual year.